src/cache_cf.cc

```diff
@@ -1490,16 +1491,12 @@ free_SBufList(SBufList *list)
 static void
 dump_acl(StoreEntry * entry, const char *name, ACL * ae)
 {
+    PackableStream os(*entry);
     while (ae != nullptr) {
         debugs(3, 3, "dump_acl: " << name << " " << ae->name);
-        storeAppendPrintf(entry, "%s %s %s ",
-                          name,
-                          ae->name,
-                          ae->typeString());
-        SBufList tail;
-        tail.splice(tail.end(), ae->dumpOptions());
-        tail.splice(tail.end(), ae->dump()); // ACL parameters
-        dump_SBufList(entry, tail);
+        os << name << " " << ae->name << " " << ae->typeString() << " "
+           << AsList(ae->dumpOptions()).delimitedBy(" ") << " "
+           << AsList(ae->dump()).delimitedBy(" ") << "\n";
```

Comment:
The above printed an extra space for ACLs without options and and extra space for ACLs without data parameters. I fixed that in PR branch commit 3df4b49. That commit also got rid of ACL::dumpOptions() that was using std::stream to build an SBufList that AsList() in this PR code was then dumping back to std::stream.